addons/auth_password_pwn_checker/models/res_users.py

```diff
@@ -0,0 +1,70 @@
+import hashlib
+import logging
+from datetime import timedelta, date
+
+import requests
+
+from odoo import _, models, api, fields
+
+_logger = logging.getLogger(__name__)
+
+
+class ResUsers(models.Model):
+    _inherit = ['res.users']
+
+    pwn_last_checked = fields.Date()
+
+    @classmethod
+    def authenticate(cls, db, login, password, user_agent_env):
+        uid = super(ResUsers, cls).authenticate(db, login, password, user_agent_env)
+        cls._check_user_password(uid, password)
+        return uid
+
+    @classmethod
+    def _check_user_password(cls, uid: int, password: str):
+        with cls.pool.cursor() as cr:
+            env = api.Environment(cr, uid, {})
+            should_check = (not env.user.pwn_last_checked or
+                            env.user.pwn_last_checked < date.today() - timedelta(days=7))
+            # if not should_check:
+            #     return
+
+            try:
+                pwned = cls._is_password_pwned(password)
+            except requests.exceptions.RequestException:
+                _logger.warning("Request to api.pwnedpasswords.com failed.")
+                return
+
+            env.user.pwn_last_checked = date.today()
+
+            if pwned:
+                body_html = _("Your password has been exposed in data breaches. Please change it immediately.")
+
+                body_html = env['mail.render.mixin']._render_encapsulate(
+                    'mail.mail_notification_light',
+                    body_html,
+                    add_context={
+                        # the 'mail_notification_light' expects a mail.message 'message' context, let's give it one
+                        'message': env['mail.message'].sudo().new(dict(body=body_html, record_name=env.user.name)),
+                        'model_description': _('Account'),
+                        'company': env.user.company_id,
+                    },
+                )
+
+                odoobot_email = env.ref('base.partner_root').email_formatted
+
+                env['mail.mail'].sudo().create({
+                    'subject': _("Your password is exposed on the Internet !"),
+                    'body_html': body_html,
+                    'email_from': odoobot_email,
+                    'email_to': env.user.email,
+                }).send()
```

Comment:
Beware send could raise some stuff, do you want to do something specific with it or not ?